web/proxy.js

```diff
@@ -24,13 +21,36 @@ console.log(`Creating proxy with host: ${host}`);
  * environment that has no local API.
  */
 const server = http.createServer((request, response) => {
+    let hostname = host;
+    let requestPath = request.url;
+
+    /**
+     * When a request is matching a proxy config path we might direct it to a different host (e.g. staging)
+     * For requests matching proxy config patterns we replace the mapping url (prefix) with the actual path.
+     * This is done because the staging api root is only intended for the proxy,
+     * the actual server request must use the /api path.
+     * For example,
+     * /api?command=OpenReport => request sent to production server
+     * /staging-api?command=OpenReport => request sent to staging server
+     * /staging-secure-api?command=OpenReport => request sent to secure staging server
+     * /chat-attachments/46545... => request sent to production server
+     * /staging-chat-attachments/46545... => request sent to staging server
```

Comment:
Related to my previous comment, if we use a slash `/` instead of `-` it makes debugging/tracing easier 

```suggestion
     * /api?command=OpenReport => request sent to production server
     * /staging/api?command=OpenReport => request sent to staging server
     * /staging-secure/api?command=OpenReport => request sent to secure staging server
     * /chat-attachments/46545... => request sent to production server
     * /staging/chat-attachments/46545... => request sent to staging server
```